### ISSUE TITLE
"newline-per-chained-call" doesn't allow first method to start on new line

### ISSUE THREAD

**Issue by PassTheMayo** [eslint/eslint#10977]
**What rule do you want to change?** I want to change the `newline-per-chained-call` rule to allow for `ignoreChainWithDepth` to be set to `0`, so every new method chained is on a new line instead of skipping the first method.

**Does this change cause the rule to produce more or fewer warnings?** The rule will produce fewer warnings by adhering to user's preference of coding style.

**How will the change be implemented? (New option, new default behavior, etc.)?** Basically you just need to change the `ignoreChainWithDepth` option to allow for `0` instead of only >= 1.

**Please provide some example code that this change will affect:**

Input:
```js
myPromise.then(() => {

}).catch(() => {

});

myPromiseSecond.get(documentID).insert({ test: true }).run().then(() => {

}).catch(() => {

});

myPromise.myObject.myProperty.get(documentID).insert({ test: true }).run().then(() => {

}).catch(() => {

});
```

Expected output:
```js
myPromise
	.then(() => {

	})
	.catch(() => {

	});

myPromiseSecond
	.get(documentID)
	.insert({ test: true })
	.run()
	.then(() => {

	})
	.catch(() => {

	});

myPromise.myObject.myProperty
	.get(documentID)
	.insert({ test: true })
	.run()
	.then(() => {

	})
	.catch(() => {

	});
```

Actual output:
```js
myPromise.then(() => {

})
	.catch(() => {

	});

myPromiseSecond.get(documentID)
	.insert({ test: true })
	.run()
	.then(() => {

	})
	.catch(() => {

	});

myPromise.myObject.myProperty.get(documentID)
	.insert({ test: true })
	.run()
	.then(() => {

	})
	.catch(() => {

	});
```

**What does the rule currently do for this code?** The `ignoreChainWithDepth` rule is only allowed to range from 1 to 10. I think this number should be allowed to be set from 0-10, allowing for the first method to be put on a new line instead of being on the same line as the variable/object it's being called from.

**What will the rule do after it's changed?** I've included examples above to what it will look like.

**Comment by platinumazure** [comment]
Hi @PassTheMayo, thanks for the issue.

Kind of unfortunate that there's an example in the docs that looks like exactly what you want, but the rule doesn't support it at this time. :frown:

I think this is a reasonable enhancement request. I'll champion this with the team.

**Comment by PassTheMayo** [comment]
Alright, thanks so much!

**Comment by PassTheMayo** [comment]
@platinumazure Also I forgot to mention something, I think there should also be a way to not force every single method to new line. I have some methods that do `this.bot.myMethod()` which I want to keep on one line, but force all methods that have been chained to a new line if that makes sense.

**Comment by kaicataldo** [comment]
Friendly ping! Where are we with this issue?

**Comment by mdjermanovic** [comment]
I think that the proposal in its current form isn't in line with how the rule works.

`newline-per-chained-call` does allow first method to start on a new line, so the title is wrong. This rule never disallows newlines. 

Proposed change can only produce more warnings, not fewer.

Setting `ignoreChainWithDepth` to `0` would require all method calls to have newline:

```js
a.b(); // bad

a
    .b(); // good
```

This probably isn't the desired behavior?

My guess is that this issue is a request for an additional option to require newlines for all methods in the chain (not just those above `ignoreChainWithDepth`) when the total depth is  `> ignoreChainWithDepth`

**Comment by bendulum** [comment]
> My guess is that this issue is a request for an additional option to require newlines for all methods in the chain (not just those above ignoreChainWithDepth) when the total depth is > ignoreChainWithDepth

I've been looking for a rule or plugin to achieve exactly this. Would be very nice to have.

Example, with option `ignoreChainWithDepth: 2`:

```js
const a = foo.bar().baz();
// this is ok

const b = foo.bar().baz().boo();
// this is too long an results in:

const b = foo.bar().baz()
    .boo();
// this is ugly and it would be better to have:

const b = foo
    .bar()
    .baz()
    .boo();
```
Maybe a new option like `preferVerticalChainWhenExceeding: true`?
